backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/lsm/LiveMigrateDiskCommand.java

```diff
@@ -308,8 +308,7 @@ private void removeImage(Guid storageDomainId, Guid imageGroupId, Guid imageId,
 
     private boolean isRemoveAutoGeneratedSnapshotRequired() {
         LiveDiskMigrateStage liveDiskMigrateStage = getParameters().getLiveDiskMigrateStage();
-        boolean removeSnapshotRequired = liveDiskMigrateStage == LiveDiskMigrateStage.LIVE_MIGRATE_DISK_EXEC_COMPLETED &&
-                liveDiskMigrateStage != LiveDiskMigrateStage.AUTO_GENERATED_SNAPSHOT_REMOVE_END;
+        boolean removeSnapshotRequired = liveDiskMigrateStage == LiveDiskMigrateStage.LIVE_MIGRATE_DISK_EXEC_COMPLETED;
```

Comment:
in my opinion this is only necessary change in this PR, the others provide no clear benefit

Reply:
The whole idea was to change the code to a simple and straightforward `switch` "state machine" instead of multiple confusing (IMHO) "if"s.
In backup and other places `performNextOperation()` is implemented using `switch` and it's really simplifies the understanding of the flow.
It took me some time to understand that actually the old code (that even calls a method that actually handles one of the `LiveDiskMigrateStage`s) is the same as the new one.

Reply:
what is the additional cognitive load of checking an `if` versus a `case`?

Reply:
If it's a single "if" then there is a small/no additional value.
But when there are multiple "if"s and you need to devote some time to understand that actually **all** of them are testing for the same enum phase/stage and even the call to some boolean method is actually another enum phase/stage handling, then it's in my opinion it has an additional value.
Assuming that I didn't enter any bugs by this refactoring, I think it's enough to look on the left side and on the right side to see that the new code is really simple and doesn't require any effort in understanding the general flow of the `performNextOperation()`:
stage1 --> stage1 handling + exit.
stage2 --> stage2 handling + exit.
...
But I will not fight for it. This was a suggestion, in my opinion it really improves the code. If there is no real value, I will drop this PR.

Reply:
ok, so once you looked for a couple of seconds and realized the `if`s are testing the phase (a common pattern in performNextOperation), why is it a problem then?

I don't think one is worse than the other, however this command always used this pattern and I see no benefit in introducing risk for such small and opinionated benefit

Reply:
I would even say that Arik's question about `AUTO_GENERATED_SNAPSHOT_REMOVE_END` in another conversation thread only proves my point that the old code was confusing enough, for example it looked like there is some handling related to `AUTO_GENERATED_SNAPSHOT_REMOVE_END` when actually it was a dead (unreachable) code. And it took me some time to understand that when I was looking at it, running flows with the help of debugger, etc.